url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/client/io/TestClientNonPPNServer.java

```diff
@@ -0,0 +1,174 @@
+import ping.pong.net.connection.DisconnectInfo;
+import ping.pong.net.connection.config.ConnectionConfigFactory;
+import ping.pong.net.connection.config.ConnectionConfiguration;
+import ping.pong.net.connection.io.ReadFullyDataReader;
+import ping.pong.net.connection.io.WriteByteArrayDataWriter;
+import ping.pong.net.connection.messaging.EnvelopeFactory;
+import ping.pong.net.connection.messaging.MessageListener;
+import ping.pong.net.server.Server;
+import ping.pong.net.server.ServerConnectionListener;
+import ping.pong.net.server.io.IoServer;
+
+/**
+ *
+ * @author mfullen
+ */
+public class TestClientNonPPNServer
+{
+    public static void main(String[] args) throws InterruptedException
+    {
+        Runnable server = new Runnable()
+        {
+            @Override
+            public void run()
+            {
+                launchPPNServer();
+                // launchNonPPNServer();
+
+            }
+        };
+
+        new Thread(server).start();
+
+
+        launchClient();
+    }
+
+    public static void launchPPNServer()
+    {
+        ConnectionConfiguration createConnectionConfiguration = ConnectionConfigFactory.createPPNConfig("localhost", 5111, 5666, false);
+        IoServer<byte[]> server = new IoServer<byte[]>(createConnectionConfiguration);
+        server.setCustomDataReader(new ReadFullyDataReader());
+        server.setCustomDataWriter(new WriteByteArrayDataWriter());
+        server.addConnectionListener(new ServerConnectionListener()
+        {
+            @Override
+            public void connectionAdded(Server server, Connection conn)
+            {
+                System.out.println("Connection Added");
+                //server.broadcast(EnvelopeFactory.createTcpEnvelope("Test"));
+            }
+
+            @Override
+            public void connectionRemoved(Server server, Connection conn)
+            {
+                System.out.println("Connection removed was " + conn.getConnectionId());
+            }
+        });
+        server.addMessageListener(new MessageListener<Connection, byte[]>()
+        {
+            @Override
+            public void messageReceived(Connection source, byte[] message)
+            {
+                System.out.println("Message Received On Server From: " + source.getConnectionId());
+                System.out.println("Message: " + new String(message));
+            }
+        });
+
+        server.start();
+    }
+
+    public static void launchNonPPNServer()
+    {
+        ServerSocketFactory socketFactory = ServerSocketFactory.getDefault();
+        ServerSocket tcpServerSocket = null;
+        try
+        {
+            tcpServerSocket = socketFactory.createServerSocket(5111);
+            tcpServerSocket.setReuseAddress(true);
+        }
+        catch (IOException ex)
+        {
+            ex.printStackTrace();
+        }
+
+
+        Socket acceptingSocket = null;
+        try
+        {
+            acceptingSocket = tcpServerSocket.accept();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *acceptingSocket*. Execution paths that do not contain closure statements were detected. Ensure *acceptingSocket* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).